From 08326b3b1092600cb67e0acb83e9eae11be54a9e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:43 -0400
Subject: [tools] perf/timechart: Add option to limit number of tasks

Message-id: <1408953210-26343-36-git-send-email-jolsa@redhat.com>
Patchwork-id: 88747
O-Subject: [PATCH RHEL7.1 BZ1133083 035/322] perf timechart: Add option to limit number of tasks
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 54874e3236b834064943c02a647823ab5d97be57
Author: Stanislav Fomichev <stfomichev@yandex-team.ru>
Date: Fri Nov 1 20:25:46 2013 +0400

description
===========
Add -n option to specify min. number of tasks to print.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Documentation/perf-timechart.txt b/tools/perf/Documentation/perf-timechart.txt
index 3ff8bd4..4373144 100644
--- a/tools/perf/Documentation/perf-timechart.txt
+++ b/tools/perf/Documentation/perf-timechart.txt
@@ -54,6 +54,10 @@ $ perf timechart
 
   Written 10.2 seconds of trace to output.svg.
 
+-n::
+--proc-num::
+        Print task info for at least given number of tasks.
+
 SEE ALSO
 --------
 linkperf:perf-record[1]
diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index bb21e57..c352be4 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -41,6 +41,7 @@
 #define SUPPORT_OLD_POWER_EVENTS 1
 #define PWR_EVENT_EXIT -1
 
+static int proc_num = 15;
 
 static unsigned int numcpus;
 static u64  min_freq; /* Lowest CPU frequency seen */
@@ -955,7 +956,7 @@ static void write_svg_file(const char *filename)
  do {
   count = determine_display_tasks(thresh);
   thresh /= 10;
- } while (!process_filter && thresh && count < 15);
+ } while (!process_filter && thresh && count < proc_num);
 
  open_svg(filename, numcpus, count, first_time, last_time);
 
@@ -1102,6 +1103,8 @@ int cmd_timechart(int argc, const char **argv,
          parse_process),
  OPT_STRING(0, "symfs", &symbol_conf.symfs, "directory",
       "Look for files with symbols relative to this directory"),
+ OPT_INTEGER('n', "proc-num", &proc_num,
+      "min. number of tasks to print"),
  OPT_END()
  };
  const char * const timechart_usage[] = {
-- 
1.7.1